lisa/nic.py

```diff
@@ -185,7 +185,21 @@ def get_nic_by_index(self, index: int = -1) -> NicInfo:
         # when there are more than one nic on the system
         number_of_nics = len(self.get_upper_nics())
         assert_that(number_of_nics).is_greater_than(0)
-        return self.nics[self.get_upper_nics()[index]]
+        try:
```

Comment:
It looks the first one capture the exception on first line, and second capture on second line. To improve readability, it's better to try/except for each line. Ideally, the try part should be as short as possible.

Reply:
@squirrelsc does the current version look good?